operators/go/submariner_controller.go

```diff
@@ -0,0 +1,221 @@
+}
+
+// newPodForCR returns a submariner pod with the same fields as the cr
+func newPodForCR(cr *submarinerv1alpha1.Submariner) *corev1.Pod {
+	labels := map[string]string{
+		"app": "submariner-engine",
+	}
+
+  // Create EnvVars for Pod
+  // TODO: Break this out into dedicated function
+  subm_namespace_env_var := corev1.EnvVar{Name: "SUBMARINER_NAMESPACE", Value: cr.Spec.SubmarinerNamespace}
+  subm_clustercidr_env_var := corev1.EnvVar{Name: "SUBMARINER_CLUSTERCIDR", Value: cr.Spec.SubmarinerClustercidr}
+  subm_servicecidr_env_var := corev1.EnvVar{Name: "SUBMARINER_SERVICECIDR", Value: cr.Spec.SubmarinerServicecidr}
+  subm_token_env_var := corev1.EnvVar{Name: "SUBMARINER_TOKEN", Value: cr.Spec.SubmarinerToken}
+  subm_clusterid_env_var := corev1.EnvVar{Name: "SUBMARINER_CLUSTERID", Value: cr.Spec.SubmarinerClusterid}
+  subm_colorcodes_env_var := corev1.EnvVar{Name: "SUBMARINER_COLORCODES", Value: cr.Spec.SubmarinerColorcodes}
+  subm_debug_env_var := corev1.EnvVar{Name: "SUBMARINER_DEBUG", Value: cr.Spec.SubmarinerDebug}
+  subm_natenabled_env_var := corev1.EnvVar{Name: "SUBMARINER_NATENABLED", Value: cr.Spec.SubmarinerNatenabled}
+  subm_broker_env_var := corev1.EnvVar{Name: "SUBMARINER_BROKER", Value: cr.Spec.SubmarinerBroker}
+  broker_k8s_apiserver_env_var := corev1.EnvVar{Name: "BROKER_K8S_APISERVER", Value: cr.Spec.BrokerK8sApiserver}
+  broker_k8s_apiservertoken_env_var := corev1.EnvVar{Name: "BROKER_K8S_APISERVERTOKEN", Value: cr.Spec.BrokerK8sApiservertoken}
+  broker_k8s_remotenamespace_env_var := corev1.EnvVar{Name: "BROKER_K8S_REMOTENAMESPACE", Value: cr.Spec.BrokerK8sRemotenamespace}
+  broker_k8s_ca_env_var := corev1.EnvVar{Name: "BROKER_K8S_CA", Value: cr.Spec.BrokerK8sCa}
+  ce_ipsec_psk_env_var := corev1.EnvVar{Name: "CE_IPSEC_PSK", Value: cr.Spec.CeIpsecPsk}
+  ce_ipsec_debug_env_var := corev1.EnvVar{Name: "CE_IPSEC_DEBUG", Value: cr.Spec.CeIpsecDebug}
+
+  // Create SecurityContext for Pod
+  security_context_add_net_admin := corev1.SecurityContext{Capabilities: &corev1.Capabilities{Add: []corev1.Capability{"NET_ADMIN"}}}
+
+  // Create Pod
+	return &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Name + "-pod",
+			Namespace: cr.Namespace,
+			Labels:    labels,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:    "submariner",
+          // TODO: Use var here
+					Image:   "submariner:local",
+          // TODO: Use var here
+					Command: []string{"submariner.sh"},
+          SecurityContext: &security_context_add_net_admin,
+          Env:     []corev1.EnvVar{subm_namespace_env_var, subm_clustercidr_env_var, subm_servicecidr_env_var, subm_token_env_var, subm_clusterid_env_var, subm_colorcodes_env_var, subm_debug_env_var, subm_natenabled_env_var, subm_broker_env_var, broker_k8s_apiserver_env_var, broker_k8s_apiservertoken_env_var, broker_k8s_remotenamespace_env_var, broker_k8s_ca_env_var, ce_ipsec_psk_env_var, ce_ipsec_debug_env_var},
```

Comment:
you can add multiple labels like this:

https://github.com/submariner-io/submariner/blob/master/test/e2e/framework/network_pods.go#L37

Easier to maintain and to read :-)

Reply:
Awesome, thanks - done :+1: